src/main/java/http_server/ResourceHandler.java

```diff
@@ -58,4 +61,30 @@ public void delete(String resourcePath) {
         );
     }
 
+    public boolean updateJSON(String resourcePath, String patchDocument) {
+        if (!patchDocument.startsWith("[")) {
+            System.err.println("Invalid JSON patch document. Terminating update request.");
+            return false;
+        }
+
+        String jsonFile = new String(router.getRepository().readFile(fullDirectoryPath + resourcePath + ".json"));
+        JSONArray patchInstructions = JSONHandler.parseArray(patchDocument);
+        JSONObject updatedJSONObject = JSONHandler.parse(jsonFile);
+
+        for (int i = 0; i < patchInstructions.length(); i++) {
+            JSONObject instructions = (JSONObject) patchInstructions.get(i);
```

Comment:
What prompted the type coercion here and below for strings?  Is that just a quirk of `get()`?

Reply:
the JSON library method of `get()` always returns an object, so I had used type conversion to force it to be the right type.

In this case, I know what the type should be so I'll update the library provided methods of `getString()` and `getJSONObject()` as appropriate